tests/hil/tests/ota/test.c

```diff
@@ -168,6 +172,24 @@ static void test_block_ops(void)
     }
 }
 
+static void test_restart(void)
+{
+
+    if (golioth_sys_sem_take(restart_test_begin_sem, 0))
```

Comment:
I don't think we need to use semaphores inside of this function. We're not performing mutual exclusion between multiple threads, and we're not signaling from one thread to another.

Reply:
Sounds good, I've converted this to use a bool variable.

Reply:
This feels a lot more straightforward!